src/components/layout/Sidebar/index.jsx

```diff
@@ -9,32 +9,36 @@ export default class Sidebar extends React.Component {
     super(props);
 
     this.state = {
-      opened: true
+      open: props.sidebar.open
     };
 
     this.triggerToggle = this.triggerToggle.bind(this);
   }
 
+  componentWillReceiveProps(nextProps) {
+    this.setState({ open: nextProps.sidebar.open });
+  }
+
   /**
    * UI EVENTS
    * - triggerToggle
   */
   triggerToggle() {
-    this.setState({
-      opened: !this.state.opened
-    });
+    const width = this.sidebarNode.offsetWidth;
+    const open = !this.state.open;
+    this.props.setSidebar(open, width);
```

Comment:
It would be better if you pass an object instead of different params and then the setSidebar function would be an `Object.assign`. What do you think?